authors/apps/articles/views.py

```diff
@@ -0,0 +1,104 @@
+from django.urls import reverse
+from django.utils.encoding import force_bytes
+from django.utils.encoding import force_text
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from rest_framework import status
+from rest_framework.generics import GenericAPIView, CreateAPIView, UpdateAPIView
+from rest_framework.generics import RetrieveUpdateAPIView, RetrieveUpdateDestroyAPIView
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.exceptions import NotFound
+
+from config.settings import default
+from .models import Article
+from .renderers import ArticleJSONRenderer
+from .serializers import (
+     ArticleSerializer
+)
+
+from authors.apps.profiles.models import Profile
+from authors.apps.authentication.models import User
+
+
+class ArticleView(APIView):
+    """creating, viewing , deleting and updating articles"""
+
+    permission_classes = (IsAuthenticated,)
+
+
+    def post(self, request):
+        serializer_context = {'author': request.user.profile}
+        article = request.data.get('article', {})
+
+        # Create an article from the above data
+        serializer = ArticleSerializer(data=article, context=serializer_context)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        return Response({"articles": serializer.data}, status=status.HTTP_201_CREATED)
+
+
+    def get(self, request):
+        queryset = Article.objects.all()
+        # the many param informs the serializer that it will be serializing more than a single article.
+        serializer = ArticleSerializer(queryset, many=True)
+        return Response({"articles": serializer.data})
+
+class ArticleRetrieveUpdateDelete(RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ArticleSerializer
+    queryset = Article.objects.all()
+    lookup_field = 'slug'
+
+
+    def retrieve(self, request, slug):
+        try:
+            serializer_instance = self.queryset.get(slug=slug)
+        except Article.DoesNotExist:
+            raise NotFound('No article found with this slug.')
+
+        serializer = self.serializer_class(serializer_instance)
+        return Response({"articles": serializer.data})
+
+
+    def destroy(self, request, slug):
+        try:
+            serializer_instance = self.queryset.get(slug=slug)
+            requester = Profile.objects.get(user=request.user)
+            is_author = serializer_instance.author == requester
+            if not is_author:
+                return Response({"response": "you can't delete this article"},
+                            status=status.HTTP_403_FORBIDDEN)
+        except Article.DoesNotExist:
+            raise NotFound('No article found with this slug.')
+
+        self.perform_destroy(serializer_instance)
+        return Response({"message":"Article has been deleted"},
```

Comment:
@frankopkusianwar , instead of creating a response json, add a `renderer class` to the view